contract_manager/scripts/deploy_evm_pulse_contracts.ts

```diff
@@ -0,0 +1,194 @@
+}
+
+const CACHE_FILE = ".cache-deploy-evm-pulse-contracts";
+
+const parser = yargs(hideBin(process.argv))
+  .scriptName("deploy_evm_pulse_contracts.ts")
+  .usage(
+    "Usage: $0 --std-output-dir <path/to/std-output-dir/> --private-key <private-key> --chain <chain> --default-provider <default-provider> --wormhole-addr <wormhole-addr>"
+  )
+  .options({
+    ...COMMON_DEPLOY_OPTIONS,
+    chain: {
+      type: "string",
+      demandOption: true,
+      desc: "Chain to upload the contract on. Can be one of the evm chains available in the store",
+    },
+    "default-provider": {
+      type: "string",
+      desc: "Address of the default provider for the Pulse contract",
+    },
+  });
+
+async function deployPulseContracts(
+  chain: EvmChain,
+  config: DeploymentConfig,
+  executorAddr: string
+): Promise<string> {
+  console.log("Deploying PulseUpgradeable on", chain.getId(), "...");
+
+  // Get the artifact and ensure bytecode is properly formatted
+  const pulseArtifact = JSON.parse(
+    fs.readFileSync(
+      path.join(config.jsonOutputDir, "PulseUpgradeable.json"),
+      "utf8"
+    )
+  );
+  console.log("PulseArtifact bytecode type:", typeof pulseArtifact.bytecode);
+
+  const pulseImplAddr = await deployIfNotCached(
+    CACHE_FILE,
+    chain,
+    config,
+    "PulseUpgradeable",
+    []
+  );
+
+  console.log("PulseUpgradeable implementation deployed at:", pulseImplAddr);
+
+  const pulseImplContract = getWeb3Contract(
+    config.jsonOutputDir,
+    "PulseUpgradeable",
+    pulseImplAddr
+  );
+
+  // Get CLI arguments for initialization
+  const argv = await parser.argv;
+
+  console.log("Preparing initialization data...");
+  console.log("Using default provider:", argv["default-provider"]);
+
+  const pulseInitData = pulseImplContract.methods
+    .initialize(
+      executorAddr, // owner
+      executorAddr, // admin
+      "1", // pythFeeInWei
+      executorAddr, // pythAddress - using executor as a placeholder
+      argv["default-provider"], // defaultProvider
+      true, // prefillRequestStorage
+      3600 // exclusivityPeriodSeconds - 1 hour
+    )
+    .encodeABI();
+
+  console.log("Deploying ERC1967Proxy for Pulse...");
+
+  // Custom deployment for ERC1967Proxy using the correct path
+  const proxyArtifactPath = path.join(
+    process.cwd(),
+    "../target_chains/ethereum/contracts/artifacts/@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol/ERC1967Proxy.json"
+  );
+
+  console.log("Loading proxy artifact from:", proxyArtifactPath);
+  const proxyArtifact = JSON.parse(fs.readFileSync(proxyArtifactPath, "utf8"));
+
+  // Handle bytecode which can be either a string or an object with an 'object' property
+  let bytecode = proxyArtifact.bytecode;
+  if (
+    typeof bytecode === "object" &&
+    bytecode !== null &&
+    "object" in bytecode
+  ) {
+    bytecode = bytecode.object;
+  }
+
+  // Ensure bytecode starts with 0x
+  if (!bytecode.startsWith("0x")) {
+    bytecode = `0x${bytecode}`;
+  }
```

Comment:
i suspect these are because you are loading a pre-built binary. but nevertheless these are now in deployifnotcached